packages/airnode-node/src/evm/handlers/process-transactions.test.ts

```diff
@@ -18,114 +19,184 @@ mockEthers({
-    const gasPrice = ethers.BigNumber.from(1000);
-    const gasPriceSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getGasPrice');
-    gasPriceSpy.mockResolvedValueOnce(gasPrice);
-
-    const balanceSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getBalance');
-    balanceSpy.mockResolvedValueOnce(ethers.BigNumber.from(250_000_000));
-
-    estimateGasWithdrawalMock.mockResolvedValueOnce(ethers.BigNumber.from(50_000));
-    fulfillWithdrawalMock.mockResolvedValueOnce({
-      hash: '0xcbb3f9dc6a24e8b6f5427dcf960b1da01c3df0636cb25a292f8dcaad78755c8d',
-    });
-    staticFulfillMock.mockResolvedValueOnce({ callSuccess: true });
-    fulfillMock.mockResolvedValueOnce({
-      hash: '0xad33fe94de7294c6ab461325828276185dff6fed92c54b15ac039c6160d2bac3',
-    });
-
-    const apiCall = fixtures.requests.buildSubmittableApiCall({
-      sponsorAddress: '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181',
-    });
-    const withdrawal = fixtures.requests.buildWithdrawal({
-      sponsorAddress: '0x99bd3a5A045066F1CEf37A0A952DFa87Af9D898E',
-    });
-    const requests: GroupedRequests = {
-      apiCalls: [apiCall],
-      withdrawals: [withdrawal],
-    };
-    const transactionCountsBySponsorAddress = {
-      '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181': 79,
-      '0x99bd3a5A045066F1CEf37A0A952DFa87Af9D898E': 212,
-    };
-    const state = fixtures.buildEVMProviderState({
-      requests,
-      transactionCountsBySponsorAddress,
-    });
-
-    const res = await processTransactions(state);
-    expect(res.requests.apiCalls[0]).toEqual({
-      ...apiCall,
-      nonce: 79,
-      fulfillment: { hash: '0xad33fe94de7294c6ab461325828276185dff6fed92c54b15ac039c6160d2bac3' },
-      status: RequestStatus.Submitted,
-    });
-    expect(res.requests.withdrawals[0]).toEqual({
-      ...withdrawal,
-      nonce: 212,
-      fulfillment: { hash: '0xcbb3f9dc6a24e8b6f5427dcf960b1da01c3df0636cb25a292f8dcaad78755c8d' },
-      status: RequestStatus.Submitted,
-    });
-    expect(res.gasTarget).toEqual({ gasPrice: ethers.BigNumber.from('1000') });
-
-    // Withdrawal was submitted
-    expect(fulfillWithdrawalMock).toHaveBeenCalledTimes(1);
-    expect(fulfillWithdrawalMock).toHaveBeenCalledWith(
-      withdrawal.id,
-      withdrawal.airnodeAddress,
-      withdrawal.sponsorAddress,
-      {
-        gasPrice,
-        gasLimit: ethers.BigNumber.from(70_000),
+  test.each(['1', '2'])(
+    'fetches the gas price, assigns nonces and submits transactions - txType: %d',
+    async (txType) => {
+      const initialConfig = fixtures.buildConfig();
+      const chainOptions = { txType } as ChainOptions;
+      const chains = initialConfig.chains.map((chain) => ({
+        ...chain,
+        chainOptions,
+      }));
+      const config = {
+        ...initialConfig,
+        chains,
+      } as Config;
+      jest.spyOn(fs, 'readFileSync').mockReturnValue(JSON.stringify(config));
```

Comment:
haha, wow, yeah, the lack of logging caused me a lot of problems debugging. I eventually wrote a file logger and then `tail -f`'ed the output to see what was going on.

Reply:
I checked further, I don't think it's because of this spy; the yarn test command sets an `env` of `SILENCE_LOGGER` which then causes an early return from the `log` function in the logger.

In `packages/airnode-node/src/logger.ts` line 45 onwards:
```
export function log(level: LogLevel, message: string, options: LogOptions) {
  if (process.env.SILENCE_LOGGER) {
    return;
  }
```

Reply:
Well, when you debug you run the tests with `SILENCE_LOGGER` disabled.